app/src/main/java/com/github/polybooks/core/database/interfaces/SaleDatabase.kt

```diff
@@ -35,15 +30,26 @@ interface SaleDatabase {
 
     /**
      * Add the given sale to the database
-     * @param sale The sale to insert
+     * @param book the isbn of the book being sold
+     * @param seller the user selling the book (can't be the local user)
+     * @param price the price of the sale
+     * @param condition the condition of the book
+     * @param state the state of the sale
+     * @param image the image describing the book being sold
      */
-    fun addSale(sale: Sale) : Unit
+    fun addSale(book : ISBN,
+                seller : User,
+                price : Float,
+                condition : BookCondition,
+                state : SaleState,
+                image : Image?) : CompletableFuture<Sale>
 
```

Comment:
As Cirrus said, there may be too much arguments here. I see that you deleted the addSale with argument of type "Sale", is there a reason ? I think we should create a class sale with these attributes (book, seller etc...)